modules/gateway/upnp.go

```diff
@@ -109,15 +109,22 @@ func (g *Gateway) threadedForwardPort(port string) {
 	}
 
 	g.log.Println("INFO: successfully forwarded port", port)
+
+	// Establish port-clearing at shutdown.
+	g.threads.AfterStop(func() {
+		g.managedClearPort(g.myAddr.Port())
```

Comment:
It isn't safe to use `g.myAddr.Port()` as the hostname may not have been discovered yet, in which case `myAddr` will be the empty string. We used to set `myAddr` to ":port", but we changed this behavior when we cleaned up the `NetAddress` and implemented better `IsValid` checking.